### PR TITLE
Add suport for XLSX imports

### LINKED ISSUE #832: Data import: support for xlsx format used in data download
As a project manager, I want to be able to import data that have been downloaded from an existing project in Cadasta Platform. 

That means that we should support the _.xlsx_ format that is used when downloading data from a project in Cadasta platform, containing the _locations_, _parties_ and _relationships_ tabs. 

Not targeted for Sprint 10.

### PR DESCRIPTION
**Pull request by bjohare**
#### Fix #832 Data Import: Add support for XLSX imports
 * Add XLSX imports
* Add import of Locations or Parties separately
* Add selection of location_type field in import
* Add selection of party_type field in import
* Add suport for 1-M relations in import
### When should this PR be merged

Can be merged immediately. 

### Risks

Low risk

### Follow up actions

Adds dependency on `pandas 0.19.0`, re-provisioning required

### CODE REVIEW

File: cadasta/organization/importers/xls.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                locations, relationships, left_on='id',
+                right_on='spatial_unit_id')
+            final = pd.merge(
+                joined, parties, left_on='party_id', right_on='id')
+            drop_cols = [
+                'id_x', 'id_y', 'party_id', 'spatial_unit_id',
+                'tenure_type.label']
+            final.drop(drop_cols, inplace=True, axis=1)
+            return final.to_csv(index=False, index_label=False)
+        except Exception:
```
Comment (oliverroick) [inline cadasta/organization/importers/xls.py]: Again, we should catch generic exceptions, only those that we actually expect in the block.

File: cadasta/organization/views/default.py [hunk 2]
```diff
@@ -834,21 +848,25 @@ def done(self, form_list, **kwargs):
...
                 name=name, description=description,
                 original_file=original_file, mime_type=mime_type,
                 contributor=self.request.user, project=self.get_project())
             upload_to = getattr(resource.file.field, 'upload_to')
-            url = default_storage.save(
-                upload_to + '/' + file.name, file.read())
+            url = ''
+            while not url:
+                temp_url = upload_to + '/' + random_id() + ext
+                try:
```
Comment (oliverroick) [inline cadasta/organization/views/default.py]: Why don't you use `if not Resource.objects.filter(file__contains=temp_url).exists()` here? Saves you two lines and is easier to read...

File: cadasta/organization/importers/base.py [hunk 3]
```diff
@@ -52,8 +75,144 @@ def get_schema_attrs(self):
...
+        for attr in attributes:
+            attribute, content_type, name = attr_map.get(attr)
+            if (attribute is not None):
+                val = row[self.csv_headers.index(attr)]
+                if (not attribute.required and val == ""):
+                    continue
+                # handle select_multiple fields
+                if (attribute.attr_type.name == 'select_multiple'):
+                    val = [v.strip() for v in val.split(',')]
+                content_types[content_type][
```
Comment (bjohare) [inline cadasta/organization/importers/base.py]: Fixed.

File: cadasta/organization/importers/xls.py (hunk 1 above)
Comment (bjohare) [inline cadasta/organization/importers/xls.py]: Refactored to catch KeyError's only here.

**Pull request merged**